pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Transactions.java

```diff
@@ -39,4 +40,14 @@
      */
     CompletableFuture<List<TransactionCoordinatorStatus>> getCoordinatorStatusList();
 
+    /**
+     * Get transaction coordinator internal stats.
+     *
+     * @param coordinatorId the coordinator id
+     * @param metadata is get ledger metadata
+     *
+     * @return the future internal stats of this coordinator
+     */
+    CompletableFuture<ManagedLedgerInternalStats> getCoordinatorInternalStats(int coordinatorId, boolean metadata);
```

Comment:
it is better to return a brand new type, dedicated to Coordinator, something like `CoordinatorInternalStats`.

because they are separate domains, the ManagedLedger domain and the Transaction Coordinator.

nothing prevents us to have an internal method that creates a CoordinatorInternalStats from a ManagedLedgerInternalStats, but that's an implementation detail

Reply:
good suggestion.